lib/engine/game/g_1880/game.rb

```diff
@@ -321,6 +322,41 @@ def event_signal_end_game!
           game_end_check
         end
 
+        def event_can_buy_trains!
+          @log << "-- Event: #{EVENTS_TEXT['can_buy_trains'][1]} --"
+          @can_buy_trains = true
+        end
+
+        def force_exchange_rocket
+          return unless @rocket
+          return if @rocket.closed?
+
+          player_corps = @corporations.select { |c| c.owner == @p7.owner }
+          corps_with_slots = player_corps.select { |c| c.trains.length < train_limit(c) }
+          if player_corps.empty?
+            @rocket.close!
+          elsif corps_with_slots.empty?
+            @depot.export!
+            @rocket.close!
+          else
+            # change rocket ability into assign
+            fix_rocket_ability
+          end
+        end
+
+        def fix_rocket_ability
+          @rocket.all_abilities.dup.each { |a| @rocket.remove_ability(a) }
+
+          assign_ability = Engine::Ability::Base.new(
+            type: 'assign_corporation',
+            closed_when_used_up: true,
+            owner_type: 'player',
+            count: 1,
+            when: 'owning_player_or_turn'
```

Comment:
Its immediate

Reply:
fixed the entire assign step. removed when